sklearn/calibration.py

```diff
@@ -0,0 +1,489 @@
+        The method to use for calibration. Can be 'sigmoid' which
+        corresponds to Platt's method or 'isotonic' which is a
+        non-parameteric approach. It is not advised to use isotonic calibration
+        with too few calibration samples (<<1000) since it tends overfit. Use 
+        sigmoids (Platt's calibration) in this case.
+
+    cv : integer or cross-validation generator or "prefit", optional
+        If an integer is passed, it is the number of folds (default 3).
+        Specific cross-validation objects can be passed, see
+        sklearn.cross_validation module for the list of possible objects.
+        If "prefit" is passed, it is assumed that base_estimator has been
+        fitted already and all data is used for calibration.
+
+    References
+    ----------
+    .. [1] Obtaining calibrated probability estimates from decision trees
+           and naive Bayesian classifiers, B. Zadrozny & C. Elkan, ICML 2001
+
+    .. [2] Transforming Classifier Scores into Accurate Multiclass
+           Probability Estimates, B. Zadrozny & C. Elkan, (KDD 2002)
+
+    .. [3] Probabilistic Outputs for Support Vector Machines and Comparisons to
+           Regularized Likelihood Methods, J. Platt, (1999)
+
+    .. [4] Predicting Good Probabilities with Supervised Learning, 
+           A. Niculescu-Mizil & R. Caruana, ICML 2005
+    """
+    def __init__(self, base_estimator=GaussianNB(), method='sigmoid', cv=3):
+        self.base_estimator = base_estimator
+        self.method = method
+        self.cv = cv
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the calibrated model
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data.
+
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like, shape = [n_samples] or None
+            Sample weights. If None, then samples are equally weighted.
+
+        Returns
+        -------
+        self : object
+            Returns an instance of self.
+        """
+        X, y = check_X_y(X, y, accept_sparse=['csc', 'csr', 'coo'])
+        X, y = indexable(X, y)
+        lb = LabelBinarizer().fit(y)
+        self.classes_ = lb.classes_
+
+        self.calibrated_classifiers_ = []
+        if self.cv == "prefit":
+            calibrated_classifier = _CalibratedClassifier(self.base_estimator,
+                                                          method=self.method)
+            if sample_weight is not None:
+                calibrated_classifier.fit(X, y, sample_weight)
+            else:
+                calibrated_classifier.fit(X, y)
+            self.calibrated_classifiers_.append(calibrated_classifier)
+        else:
+            cv = _check_cv(self.cv, X, y, classifier=True)
+            for train, test in cv:
+                this_estimator = clone(self.base_estimator)
+                if sample_weight is not None and \
```

Comment:
Why is it silently ignored if it is not in argspec? because it is used in the calibration? seems slightly magic, but not sure what to do about it.